business/checkers/authorization/no_host_checker.go

```diff
@@ -49,7 +49,10 @@ func (n NoHostChecker) validateHost(ruleIdx int, to []*api_security_v1beta.Rule_
 	if len(to) == 0 {
 		return nil, true
 	}
-
+	namespace, clusterName := n.AuthorizationPolicy.Namespace, n.AuthorizationPolicy.ClusterName
```

Comment:
I was not aware of the "ClusterName" field in the ObjectMeta; I think it's there for future use according to the documentation?

![image](https://user-images.githubusercontent.com/1662329/148765054-1f85c2fd-14dd-46ba-b8c3-2ffe0a0cb44a.png)

I see in the "kubernetes.GetHost" that it is also expecting an empty value and taking the default domain as well:
![image](https://user-images.githubusercontent.com/1662329/148765321-c4e355a7-08f0-44e8-ae55-28bd72196001.png)

Perhaps it can keep passing an empty? (I'm not entirely familiar with the logic, but maybe it's not needed in another place).

Reply:
The ClusterName is used in several places of validations, when building FQDN for the service host.
But there is code inconsistency between validations, some places they can use only the default(empty) value and generate FQDN wrongly, so I am changing it to use configured cluster name when the default value is empty.